packages/Search/src/details/DTK_DetailsAlgorithms.hpp

```diff
@@ -101,10 +167,13 @@ double distance( Point const &point, Box const &box )
 
 // distance point-sphere
 KOKKOS_INLINE_FUNCTION
-double distance( Point const &point, Sphere const &sphere )
+DistanceReturnType distance( Point const &point, Sphere const &sphere )
 {
-    return KokkosHelpers::max(
+    // FIXME this triggers implicit conversion from DistanceReturnType to double
+    // which is precisely what I am trying to avoid...
+    double tmp = KokkosHelpers::max(
```

Comment:
can you explicitly cast to `DistanceReturnType` to prevent the implicit cast?

Reply:
What triggers the implicit conversion to double here is the operator- (the subtraction).  I am not sure how I want to deal with that one yet.  This is not call anywhere in the traversals selected for the benchmark.

Reply:
Do we implement a templated `operator -` to deal with this?

Reply:
> Do we implement a templated operator - to deal with this?

What do you mean?